### PR TITLE
1567 split well completion in control and view

### PR DESCRIPTION
**Pull request by jorgenherje**
Split previous `package/well-completions` package into a view-component (`well-completions-plot`), independent of `redux` and `mui`. The old `WellCompletions`-component is now placed in `packages/dash-components` for usage in Dash.

Thereby the new component named `well-completions-plot` can be utilized, making it possible to use the view only.

---

Closes: #1567

### CODE REVIEW

**Comment by jorgenherje** [comment]
Note:

Currently issues with `package-lock.json` for `typescript/`-folder, and the following `packages`: `dash-components` and `well-completions-plot`. The `package-lock.json` files are not up to date, missing correct names and are not updated according to the respective `package.json` files. Manually removed `well-completions` package from `typescript/package-lock.json` in commit `556fa39`

**Comment by jorgenherje** [comment]
Note:

How to handle .ts/.tsx files for package? As of now the main is set to `/dist/index.js` in `package.json` for `well-completions-plot`-package. Can `/src/index.ts` be used instead?

AND, how can `plotData` type be exported for usage externally?

**Comment by hkfb** [comment]
> How to handle .ts/.tsx files for package? As of now the main is set to `/dist/index.js` in `package.json` for `well-completions-plot`-package. Can `/src/index.ts` be used instead?

Probably `.js` was used initially because Dash components couldn't be generated from typescript files. `.js` files should now be generated in the transpilation step, so it should be safe and preferred to use `.ts` instead. 

> AND, how can `plotData` type be exported for usage externally?

You can find some examples of exporting types in the other packages.

File: typescript/packages/dash-components/package.json [hunk 1]
```diff
@@ -1,26 +1,32 @@
...
+    "main": "index.js",
+    "scripts": {
+        "test": "echo \"Error: no test specified\" && exit 1"
+    },
+    "author": "Equinor <opensource@equinor.com>",
+    "license": "MPL-2.0",
+    "dependencies": {
+        "@equinor/eds-core-react": "^0.12.1",
+        "@equinor/eds-icons": "^0.19.1",
+        "@mui/material": "^5.12.2",
```
Comment (hkfb) [inline typescript/packages/dash-components/package.json:14]: Suggest moving to `peerDependencies`. Consumers may want to use different potentially conflicting versions.

**Comment by hkfb** [comment]
Conventional commits are enforced now, so you will need to reformat the commit messages :-/

https://www.conventionalcommits.org/en/v1.0.0/#specification

https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Comment by jorgenherje** [comment]
> Conventional commits are enforced now, so you will need to reformat the commit messages :-/
> 
> https://www.conventionalcommits.org/en/v1.0.0/#specification
> 
> https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

I assume this is only necessary for the last commit message, as we use squash and merge here? Shouldn't be a need to update all previous messages for this branch?

**Comment by hkfb** [comment]
> > Conventional commits are enforced now, so you will need to reformat the commit messages :-/
> > https://www.conventionalcommits.org/en/v1.0.0/#specification
> > https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History
> 
> I assume this is only necessary for the last commit message, as we use squash and merge here? Shouldn't be a need to update all previous messages for this branch?

I'm afraid so ATM. I guess we could consider enforcing only PR titles.

**Comment by jorgenherje** [comment]
> > > Conventional commits are enforced now, so you will need to reformat the commit messages :-/
> > > https://www.conventionalcommits.org/en/v1.0.0/#specification
> > > https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History
> > 
> > 
> > I assume this is only necessary for the last commit message, as we use squash and merge here? Shouldn't be a need to update all previous messages for this branch?
> 
> I'm afraid so ATM. I guess we could consider enforcing only PR titles.

I feel the main branch should not be interested in messages of each commit in a working branch, especially when one performs squash commit when merging into the main branch on PR's.

So is it enough to edit the commit message and title for the PR to work with conventional commit?

**Comment by hkfb** [comment]
> I feel the main branch should not be interested in messages of each commit in a working branch, especially when one performs squash commit when merging into the main branch on PR's.

The rationale is that, if we move to semantic releases, then a PR can contain multiple changes that can trigger a release or an entry in the changelog. But I understand that it could be cumbersome. Maybe we'll find out that it's not needed after all when we get a chance to try it out.

> So is it enough to edit the commit message and title for the PR to work with conventional commit?

No, for now you will need to change all the commits in the PR or squash them. It's quite straightforward with eg. `git rebase -i upstream/master`

**Comment by jorgenherje** [comment]
> > I feel the main branch should not be interested in messages of each commit in a working branch, especially when one performs squash commit when merging into the main branch on PR's.
> 
> The rationale is that, if we move to semantic releases, then a PR can contain multiple changes that can trigger a release or an entry in the changelog. But I understand that it could be cumbersome. Maybe we'll find out that it's not needed after all when we get a chance to try it out.
> 
> > So is it enough to edit the commit message and title for the PR to work with conventional commit?
> 
> No, for now you will need to change all the commits in the PR or squash them. It's quite straightforward with eg. `git rebase -i upstream/master`

I understand. There was a lot of conflicts and a bit work to do when it came to rebasing. A lot of deleted files and moved files had been edited after my branch was created. I made a new PR with the new rebased branch: https://github.com/equinor/webviz-subsurface-components/pull/1629

**Pull request closed without merging**